pkg/services/filters.go

```diff
@@ -0,0 +1,260 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/databasemigrationservice"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/prometheusservice"
+	"github.com/aws/aws-sdk-go/service/storagegateway"
+
+	"github.com/nerdswords/yet-another-cloudwatch-exporter/pkg/config"
+	"github.com/nerdswords/yet-another-cloudwatch-exporter/pkg/model"
+	"github.com/nerdswords/yet-another-cloudwatch-exporter/pkg/promutil"
+)
+
+type serviceFilter struct {
+	// ResourceFunc can be used to fetch additional resources
+	ResourceFunc func(context.Context, TagsInterface, *config.Job, string) ([]*TaggedResource, error)
+
+	// FilterFunc can be used to the input resources or to drop based on some condition
+	FilterFunc func(context.Context, TagsInterface, []*TaggedResource) ([]*TaggedResource, error)
+}
+
+// filterFuncs maps a service namespace to (optional) serviceFilter
+var filterFuncs map[string]serviceFilter = map[string]serviceFilter{
+	"AWS/ApiGateway": {
```

Comment:
WDYT about adding a test to filters_test.go which checks that all keys produce a non-nil service when calling `config.SupportedServices.GetService(`? Might help prevent some awkward bugs later.

Reply:
+1, to keep aligned both maps

Reply:
Just to doble-check my understanding, do you mean checking that every key used here is a vaild service name as defined by `config.SupportedServices`?

Reply:
That correct, ideally it would catch if a someone was trying to apply serviceFilters to an incorrect name.